Update architecture diagram

suzannehamilton opened the pull request:
- Add EFS to the backend checks
- Add connections from backend processes to the API
- Show the antivirus lambda copying the files to the clean and qurantine buckets
- Add API Gateway to the export workflow

MancunianSam commented:
Looks good. Couple of minor things/questions

* You have the `get file` step going to the EFS store from the backend queues rather than the backend check lambdas.
* Do we want to include the export task EFS as well?
* Do we want to include the authorisation lambda for the export task. This is attached to the API gateway and calls keycloak and the graphql api? This may make the diagram a bit complicated so maybe leave it.
* Do you want to include the step function between the API gateway and the export ECS task? 

The diagram makes sense without these things so I don't mind if we leave them so things are clearer. What do you think?

suzannehamilton commented:
@MancunianSam Thanks for the feedback. I've updated the diagram:

* Fixed the file check so that the "get file" line comes from the lambdas rather than SQS
* Added the export EFS
* Added the export auth lambda, though I haven't shown the connection to Keycloak because it was getting a bit cluttered
* Added a step function box around the export task

What do you think of the new version?

I'm also thinking of splitting up the diagram so that the overall diagram shows a simplified version of the file check and export workflows, and creating more accurate diagrams for each of those separately. Though I'll probably come back to that in a separate PR, if you think this is good enough for now?

MancunianSam reviewed (approved):
That looks good now thanks. I think it would be helpful to have the simplified file checks and export tasks actually but for now this is good :+1:
